File: 137-single-number-ii/single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_map<int,int>map;
        int ans=0;
        for(auto x:nums)
            map[x]++;
        
        // for(auto it:map)
        // {
        //     if(it.second==1)
        //         return it.first;
        // }

        for(auto it:nums)
        {
            if(map[it]==1)
            {
                ans=it;
                break;
            }
                
        }
        return ans;
    }
};

File: LinkedList/easy/137-single-number-ii/single-number-ii.java
class Solution {
    public int singleNumber(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int x : nums)
            map.put(x, map.getOrDefault(x, 0) + 1);

        for (int x : nums) {
            if (map.get(x) == 1)
                return x;
        }
        return 0;
    }
}